### ISSUE TITLE
Inlay hints aren't showing at all

### ISSUE THREAD

**Issue by gptlang** [ray-x/go.nvim#499]
```
$ gopls version
golang.org/x/tools/gopls v0.16.1

$ nvim --version

NVIM v0.11.0-dev+657-gbb4b6b427
Build type: RelWithDebInfo
LuaJIT 2.1.1720049189
```

Config copied directly from readme. Nothing show sup at all

**Comment by antoineco** [comment]
This is likely caused by `hints` being defaulted to an empty table [since commit `a7468a83`](https://github.com/ray-x/go.nvim/commit/a7468a838bf3a5b7aa2fba3ba9a5e03497ba82c6#diff-901c682f3c26fd82f2c4d066ce1dd4e307e521b6dca87d197a518adae8c4bf9eL373-R385): https://github.com/ray-x/go.nvim/blob/8389adb56ab124b75d858df5f87b741eaca7b8b3/lua/go/gopls.lua#L385

I see the following error in my Neovim notifications/messages: `LSP[gopls] Invalid settings: parsing setting "hints": invalid type []interface {} for map[string]bool option`.

Does adding the option below to your gopls configuration solve the issue for you? ([reference](https://github.com/golang/tools/blob/gopls/v0.16.1/gopls/doc/settings.md#inlayhint))

```lua
require("go").setup {
  lsp_cfg = {
    settings = {
      gopls = {
        hints = {
          parameterNames = true
          -- ... other settings from doc/inlayHints.md ...
        }
      }
    }
  }
}
```

**Comment by antoineco** [comment]
Ref
- #500
- #501

**Comment by gptlang** [comment]
Thanks. That indeed does fix this issue.

**Comment by antoineco** [comment]
This can now be closed due to the merge of #501.
